### PR TITLE
Fix golangci issues on master branch

### PR DESCRIPTION
**Pull request by edwardstudy**
A chore.

Added TODO about errchecking on BindFlag.

### CODE REVIEW

File: pkg/webhook/config.go [hunk 1]
```diff
@@ -205,7 +205,10 @@ func (f *Config) CreateValidationWebhookServerConfig(ctx context.Context, webhoo
 		}
 	}
 	ctxlog.Debugf(ctx, "Creating validation webhook config '%s'", config.Name)
-	f.client.Delete(ctx, config)
+	err := f.client.Delete(ctx, config)
+	if err != nil && !apierrors.IsNotFound(err) {
```
Comment (mudler) [inline pkg/webhook/config.go:209]:
```suggestion
	if err := f.client.Delete(ctx, config); err != nil && !apierrors.IsNotFound(err) {
```

File: pkg/webhook/config.go [hunk 2]
```diff
@@ -251,7 +254,10 @@ func (f *Config) CreateMutationWebhookServerConfig(ctx context.Context, webhooks
 	}
 
 	ctxlog.Debugf(ctx, "Creating mutating webhook config '%s'", config.Name)
-	f.client.Delete(ctx, &config)
+	err := f.client.Delete(ctx, &config)
+	if err != nil && !apierrors.IsNotFound(err) {
```
Comment (mudler) [inline pkg/webhook/config.go:258]:
```suggestion
	if err := f.client.Delete(ctx, &config); err != nil && !apierrors.IsNotFound(err) {
```

File: pkg/logger/logger.go [hunk 3]
```diff
@@ -27,7 +27,10 @@ func newLogger(level string, options ...zap.Option) *zap.Logger {
 	}
 
 	l := zap.DebugLevel
-	l.Set(level)
+	err := l.Set(level)
+	if err != nil {
```
Comment (mudler) [inline pkg/logger/logger.go:31]:
```suggestion
	if err := l.Set(level); err != nil {
```

File: pkg/ctxlog/events.go [hunk 4]
```diff
@@ -112,7 +112,11 @@ func (ev Event) debugJSON(ctx context.Context, objectInfo interface{}) {
 
 	// treat JSON data as a string map and extract message
 	var result map[string]string
-	json.Unmarshal([]byte(jsonData), &result)
+	err := json.Unmarshal([]byte(jsonData), &result)
+	if err != nil {
```
Comment (mudler) [inline pkg/ctxlog/events.go:116]:
```suggestion
	if err := json.Unmarshal([]byte(jsonData), &result); err != nil {
```

File: pkg/cmd/argtoenv.go [hunk 5]
```diff
@@ -12,6 +12,7 @@ func AddEnvToUsage(cmd *cobra.Command, argToEnv map[string]string) {
 	flagSet := make(map[string]bool)
 
 	for arg, env := range argToEnv {
+		//nolint:errcheck
```
Comment (manno) [inline pkg/cmd/argtoenv.go:15]: Not sure if we want to litter our code with linter instructions.  We could ignore those in the config https://github.com/cloudfoundry-incubator/quarks-operator/commit/29e571ffbc464dec2c62cc9a259b97169ae0df65

What do you think?

File: pkg/crd/crd.go [hunk 6]
```diff
@@ -68,7 +68,7 @@ func ApplyCRD(ctx context.Context, client extv1client.ApiextensionsV1beta1Interf
 }
 
 // WaitForCRDReady blocks until the CRD is ready.
-func WaitForCRDReady(ctx context.Context, client extv1client.ApiextensionsV1beta1Interface, crdName string) error {
+func WaitForCRDReady(ctx context.Context, client extv1client.ApiextensionsV1beta1Interface, crdName string) error { // nolint:interfacer
```
Comment (manno) [inline pkg/crd/crd.go:71]: Does anyone  intend to use the output of interfacer? Maybe we drop that linter

Comment (edwardstudy) [reply pkg/crd/crd.go:71]: @manno I had some feeling. 
BTW, interface is out of maintain: https://github.com/mvdan/interfacer
> Deprecated: A tool that suggests interfaces is prone to bad suggestions, so its usefulness in real code is limited.

File: pkg/ctxlog/events.go [hunk 7]
```diff
@@ -112,7 +112,10 @@ func (ev Event) debugJSON(ctx context.Context, objectInfo interface{}) {
 
 	// treat JSON data as a string map and extract message
 	var result map[string]string
-	json.Unmarshal([]byte(jsonData), &result)
+	if err := json.Unmarshal([]byte(jsonData), &result); err != nil {
+		return
```
Comment (mudler) [inline pkg/ctxlog/events.go:116]: Maybe it would be helpful to print the error to debug if it fails?

File: pkg/crd/crd.go (hunk 6 above)
Comment (manno) [reply pkg/crd/crd.go]: Ok, added a commit to drop it

File: pkg/ctxlog/events.go (hunk 7 above)
Comment (manno) [reply pkg/ctxlog/events.go:116]: Maybe it was silent for a reason, but then a few lines done it does print a debug msg. So why not here, too.

**Pull request merged**